microprofile/connectors/kafka/src/main/java/io/helidon/microprofile/connectors/kafka/BasicKafkaConsumer.java

```diff
@@ -0,0 +1,206 @@
+ */
+
+package io.helidon.microprofile.connectors.kafka;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.helidon.common.context.Context;
+import io.helidon.common.context.Contexts;
+import io.helidon.config.Config;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * Basic Kafka consumer covering basic use-cases.
+ * Configurable by Helidon {@link io.helidon.config.Config Config},
+ * For more info about configuration see {@link KafkaConfigProperties}
+ *
+ * @param <K> Key type
+ * @param <V> Value type
+ * @see KafkaConfigProperties
+ * @see io.helidon.config.Config
+ */
+class BasicKafkaConsumer<K, V> implements Closeable {
+
+    private static final Logger LOGGER = Logger.getLogger(BasicKafkaConsumer.class.getName());
+    private static final String POOL_TIMEOUT = "pool.timeout";
+    private static final String PERIOD_EXECUTIONS = "period.executions";
+    private final KafkaConfigProperties properties;
+    private final Config config;
+    private final Lock taskLock = new ReentrantLock();
+    private final PartitionsAssignedLatch partitionsAssignedLatch = new PartitionsAssignedLatch();
+    private final List<String> topicNameList;
+    // It is not thread safe. It needs to be closed in the same thread it reads events.
+    // We need to keep the reference here to be able to wake up from pooling when shuting down
+    private final KafkaConsumer<K, V> consumer;
+    private final ScheduledExecutorService scheduler;
+
+    /**
+     * Kafka consumer created from {@link io.helidon.config.Config config}
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param config Helidon {@link io.helidon.config.Config config}
+     * @param scheduler Helidon {@link java.util.concurrent.ScheduledExecutorService scheduler}
+     */
+    BasicKafkaConsumer(Config config, ScheduledExecutorService scheduler) {
+        this.config = config;
+        this.properties = new KafkaConfigProperties(config);
+        this.topicNameList = properties.getTopicNameList();
+        this.consumer = new KafkaConsumer<>(properties);
```

Comment:
Please do not `extends Properties` in `KafkaConfigProperties`.
Add a method `toProperties` to that class that would return the properties required by `KafkaConsumer`

Reply:
I implemented it differently